### PR TITLE
And/teams features

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description 
Add the functionality to work with teams 
- A member's team can see the files that other members have uploaded .
- A Staff member get the original view where has the option to see all the uploads 
- If a member's team shares the link's file with another member of a different team, the last one will not be able to see that

## Reviewers 
- [ ] @jfavellar90 
- [ ] @diegomillan

### CODE REVIEW

File: edx_sga/api_teams.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Api for teams
+"""
+import os
+import logging
+import requests
+from oauthlib.oauth2 import LegacyApplicationClient
+from requests_oauthlib import OAuth2Session
+
+LOG = logging.getLogger(__name__)
+os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
```
Comment (jfavellar90) [inline edx_sga/api_teams.py:11]: Why you had to set this env var?

File: edx_sga/static/js/src/sga_team_view.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          element, 'get_team_grading_data'
+        );
+    var downloadUrl = runtime.handlerUrl(element, 'team_download');
+    var uploadUrl = runtime.handlerUrl(element, 'upload_assignment');
+    var finalizeUploadUrl = runtime.handlerUrl(element, 'finalize_uploaded_assignment');
+    var gradingTemplate;
+    var dataState;
+    var block;
+
+    function renderTeamGrading(state){
```
Comment (diegomillan) [inline edx_sga/static/js/src/sga_team_view.js:17]: are these functions copied from the sga original view???

Comment (andrey-canon) [reply edx_sga/static/js/src/sga_team_view.js:17]: yes, they are copied but have little modifications

**Review by diegomillan — APPROVED** [review]
Looks good to me given we already tested rocketchat api module.

File: edx_sga/sga.py [hunk 3]
```diff
@@ -1015,6 +1072,82 @@ def runtime_user_is_staff(self):
...
         return self.is_course_staff()
 
+    def team_grading_data(self):
+        """
+        Return team member assignment information for display on the
+        grading screen.
+        """
+        members = self.get_teams_members()
+        if not members:
+            return {"assignments":{}}
```
Comment (jfavellar90) [inline edx_sga/sga.py]: Should this return an empty list instead of returning an empty dict?

File: edx_sga/api_teams.py (hunk 1 above)
Comment (andrey-canon) [reply edx_sga/api_teams.py:11]: this allows to use http
 https://stackoverflow.com/questions/27785375/testing-flask-oauthlib-locally-without-https

File: edx_sga/sga.py (hunk 3 above)
Comment (andrey-canon) [reply edx_sga/sga.py]: you are right

File: edx_sga/sga.py [hunk 4]
```diff
@@ -1079,7 +1079,7 @@ def team_grading_data(self):
         """
         members = self.get_teams_members()
         if not members:
-            return {"assignments":{}}
+            return {"assignments":list()}
```
Comment (jfavellar90) [inline edx_sga/sga.py]: or [ ] ?

**Review by jfavellar90 — APPROVED** [review]

**Pull request merged**